lib/ecto/adapters/mysql/connection.ex

```diff
@@ -725,10 +732,13 @@ if Code.ensure_loaded?(Mariaex) do
       <<?`, name::binary, ?`>>
     end
 
-    defp assemble(list) do
+    defp assemble(list), do: assemble(list, " ")
+    defp assemble(list, joiner) do
       list
       |> List.flatten
-      |> Enum.join(" ")
+      |> Enum.reject(&is_nil(&1))
```

Comment:
Can we drop the is_nil check? I think removing both nil and empty strings is a smell, we should only remove the empty string ones.